File: Excel/700-799/724/724 Challenge.py
import pandas as pd
import re

path = "700-799/724/724 Reverse alphabets and numbers separately.xlsx"
input = pd.read_excel(path, usecols="A", nrows=11)
test = pd.read_excel(path, usecols="B", nrows=11)

def reverse_independently(s):
    s = str(s)
    letters = [c for c in s if c.isalpha()][::-1]
    digits = [c for c in s if c.isdigit()][::-1]
    li = iter(letters)
    di = iter(digits)
    return ''.join(next(li) if c.isalpha() else next(di) if c.isdigit() else c for c in s)

input['Reversed'] = input.iloc[:,0].map(reverse_independently)

print(input["Reversed"].equals(test['Answer Expected']))
# One row is different, but the rest are correct

File: Excel/700-799/724/724 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/700-799/724/724 Reverse alphabets and numbers separately.xlsx"
input = read_excel(path, range = "A1:A11")
test = read_excel(path, range = "B1:B11")

reverse_independently ==
  function(x) {
    chars = stringr::str_split(x, "", simplify = TRUE)
    is_letter = stringr::str_detect(chars, "[a-zA-Z]")
    is_digit = stringr::str_detect(chars, "[0-9]")
    chars[is_letter] = rev(chars[is_letter])
    chars[is_digit] = rev(chars[is_digit])
    stringr::str_c(chars, collapse = "")
  }

result = input %>%
  mutate(Reversed = map_chr(Strings, reverse_independently))

all.equal(result$Reversed, test$`Answer Expected`, check.attributes = FALSE)
# [1] "1 string mismatch"
